### ISSUE TITLE
Ansible-lint violations

### ISSUE THREAD

**Issue by brandor5** [snapp/template-ansible-collection#1]
~~~
pre-commit run --all-files
Ansible-lint.............................................................Failed
- hook id: ansible-lint
- exit code: 2

INFO     Identified /home/bsawyers/work/scratch/brandor5/cc_testing as project root due .git directory.
INFO     Set ANSIBLE_LIBRARY=/home/bsawyers/.cache/ansible-compat/6ff13f/modules:/home/bsawyers/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/bsawyers/.cache/ansible-compat/6ff13f/collections:/home/bsawyers/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/bsawyers/.cache/ansible-compat/6ff13f/roles:roles:/home/bsawyers/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
INFO     Running from /home/bsawyers/work/scratch/brandor5/cc_testing : ansible-galaxy collection install -vvv --force .
INFO     Loading ignores from .gitignore
INFO     Excluded: .git
INFO     Set ANSIBLE_LIBRARY=/home/bsawyers/.cache/ansible-compat/6ff13f/modules:/home/bsawyers/.ansible/plugins/modules:/usr/share/ansible/plugins/modules
INFO     Set ANSIBLE_COLLECTIONS_PATH=/home/bsawyers/.cache/ansible-compat/6ff13f/collections:/home/bsawyers/.ansible/collections:/usr/share/ansible/collections
INFO     Set ANSIBLE_ROLES_PATH=/home/bsawyers/.cache/ansible-compat/6ff13f/roles:roles:/home/bsawyers/.ansible/roles:/usr/share/ansible/roles:/etc/ansible/roles
WARNING  Listing 8 violation(s) that are fatal
yaml[line-length]: Line too long (269 > 160 characters)
.github/settings.yml:91

yaml[line-length]: Line too long (174 > 160 characters)
.github/settings.yml:99

yaml[line-length]: Line too long (162 > 160 characters)
.github/settings.yml:107

yaml[truthy]: Truthy value should be one of [false, true]
.github/workflows/ansible-lint.yml:6

yaml[truthy]: Truthy value should be one of [false, true]
.github/workflows/ansible-test.yml:21

yaml[truthy]: Truthy value should be one of [false, true]
.github/workflows/antsibull-docs-lint.yml:6

yaml[line-length]: Line too long (165 > 160 characters)
.github/workflows/antsibull-docs-lint.yml:12

yaml[indentation]: Wrong indentation: expected at least 1
changelogs/config.yml:12

Read documentation for instructions on how to ignore specific rule violations.

                Rule Violation Summary                
 count tag               profile rule associated tags 
     1 yaml[indentation] basic   formatting, yaml     
     4 yaml[line-length] basic   formatting, yaml     
     3 yaml[truthy]      basic   formatting, yaml     

Failed: 8 failure(s), 0 warning(s) on 23 files. Last profile that met the validation criteria was 'min'.

check for added large files..........................(no files to check)Skipped
check python ast.....................................(no files to check)Skipped
check that executables have shebangs.................(no files to check)Skipped
check json...........................................(no files to check)Skipped
check for merge conflicts............................(no files to check)Skipped
check for broken symlinks............................(no files to check)Skipped
check yaml...........................................(no files to check)Skipped
fix end of files.....................................(no files to check)Skipped
fix python encoding pragma...........................(no files to check)Skipped
mixed line ending....................................(no files to check)Skipped
pretty format json...................................(no files to check)Skipped
fix requirements.txt.................................(no files to check)Skipped
trim trailing whitespace.............................(no files to check)Skipped
rst ``code`` is two backticks........................(no files to check)Skipped
rst directives end with two colons...................(no files to check)Skipped
rst ``inline code`` next to normal text..............(no files to check)Skipped
~~~

**Comment by brandor5** [comment]
Do you have a preference of using comments to skip specific violations or skipping a rule all together?

**Comment by brandor5** [comment]
Now that I've spent some time looking at the issue and seeing what is happening, I think we disable all yaml tests inside of ansible-lint.

Then we implement a separate check for yamllint.

Thoughts?

**Comment by snapp** [comment]
Ok, so several of the lint failures we see here are with `.gitconfig/*` and were addressed in the top level `.pre-commit-config.yaml` file but weren't carried into the template contents configuration.

Note the exclusion for `check-yaml`

```
- id: check-yaml
        exclude: .*/\.github/.*
```

To answer your questions of opinion ... I prefer comments to skip violations if they are _very rarely_ bumped in to.

If we decide to address the `line-length` (even though the above exclude might silence the errors), we'd just need a solid decision on whether we care about manual line wrapping in our projects.

If we do, let's just fix the lint failures and truck on, if we don't then I'd go for skipping the rule completely.

If I were to dig in to this, my first stop would be to take a look to see if any of the following communities of practice have provided guidance

- [Ansible Community Guide](https://docs.ansible.com/ansible/devel/community/)
- [Red Hat Communities of Practice](https://github.com/redhat-cop/)
- [Ansible Validated Content Discussion](https://github.com/ansible/validated-content-discussion)

**Comment by brandor5** [comment]
I'm okay leaving the default in, however the default is different between ansible-lint and yamllint. Since this is primarily an ansible project (even though some of these violations don't occur in ansible code), I think we should enforce the default ansible line length and disable the yamllint checking of line length.

There is mention of line length in the [Automation Good Practices doc](https://redhat-cop.github.io/automation-good-practices/#_yaml_and_jinja2_syntax).

For the comments that disable rules, I think I've figured out mostly how to get rid of those without comments.

As for the top level config not getting carried down into the template directory, I don't think we should carry all of that down. It's relavent for this template repo, but most likely not for the collection the user is intending to build. I believe that should be as minimal as possible to make it work and allow the end user to form their own opinions.

**Comment by snapp** [comment]
> I'm okay leaving the default in, however the default is different between ansible-lint and yamllint. Since this is primarily an ansible project (even though some of these violations don't occur in ansible code), I think we should enforce the default ansible line length and disable the yamllint checking of line length.

sounds great, thanks for the pointer to the appropriate section in Good Practices, that's exactly what I would have wanted to refer to

> As for the top level config not getting carried down into the template directory, I don't think we should carry all of that down. It's relavent for this template repo, but most likely not for the collection the user is intending to build. I believe that should be as minimal as possible to make it work and allow the end user to form their own opinions.

I definitely don't think it should be 1-to-1 as there are certain `cookiecutter` specific configs in there, but I actually had intended this template to be a _highly opinionated_ extension of already existing project initialization approaches (e.g., `ansible-galaxy collection init` and GitHub Template projects) for Ansible Collections.

My goal was to demonstrate through tooling (e.g., linters, test pipelines, pre-commit hooks, etc.)  a high degree of hand-holding that would enable a relatively new Ansible user to get a fully functional project stamped out with best practices enforced so they could immediately `git init` -> `git add .` -> `git push`) and have a successful pipeline run in GitHub Actions.

Then they could simply focus on implementing the logic for the collection.

It would then be up to them to learn what areas they want to relax those recommendations in the long run and ideally communicate there reasoning back as issues to be discussed here.

--
Snapp

**Comment by snapp** [comment]
Resolved by #3
